### PR TITLE
#35793 Handle special characters in workspace

### PR DESCRIPTION
**Pull request by serge-rider**

### CODE REVIEW

**Review by Destrolaric — APPROVED** [review]

**Comment by Matvey16** [comment]
This fix somehow breaks even the working workspace

```
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 4: file:/C:/Users/kitsu/AppData/Roaming/DBeaverData/workspace6/
	at java.base/sun.nio.fs.WindowsPathParser.normalize(Unknown Source)
	at java.base/sun.nio.fs.WindowsPathParser.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsPathParser.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsPath.parse(Unknown Source)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(Unknown Source)
	at java.base/java.nio.file.Path.of(Unknown Source)
	at org.jkiss.dbeaver.utils.GeneralUtils.makeURIFromFilePath(GeneralUtils.java:798)
	at org.jkiss.dbeaver.utils.RuntimeUtils.getLocalFileFromURL(RuntimeUtils.java:212)
	at org.jkiss.dbeaver.utils.GeneralUtils.getMetadataFolder(GeneralUtils.java:777)
	at org.jkiss.dbeaver.core.DBeaverActivator.resetSettingsStartupActions(DBeaverActivator.java:210)
	at org.jkiss.dbeaver.core.DBeaverActivator.start(DBeaverActivator.java:89)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:833)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:825)
	... 30 more
``

**Comment by Matvey16** [comment]
Verified on Windows and MacOS

**Comment by Destrolaric** [comment]
image
The pattern will not for work for path with `//`

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/GeneralUtils.java [hunk 1]
```diff
@@ -63,6 +63,8 @@
 public class GeneralUtils {
     private static final Log log = Log.getLog(GeneralUtils.class);
 
+    public static final Pattern URI_SCHEMA_PATTERN = Pattern.compile("([a-zA-Z0-9]+:/).+");
```
Comment (Destrolaric) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/GeneralUtils.java:66]: My comment above

**Pull request merged**